### DISCUSSION TITLE
Point size

### DISCUSSION THREAD

**Discussion by MarijaHorvat** [openscenegraph/OpenSceneGraph#1315] [Q&A] [answered]
Hi,
Is there a way to use osg::PointSprite and set size of points without using osg::Point->setSize function?

**Comment by robertosfield** [comment]
You'll need to explain where you are coming from as the question is perplexing.  What are you trying to do, why and how?

**Comment by MarijaHorvat** [comment]
I am using osg in my application and I get warning  "Warning: Point::apply(State&)" when trying to use osg::Point->setSize. 

This is my CmakeLists:
    set( FETCHCONTENT_TRY_FIND_PACKAGE_MODE "NEVER" )

    set( BUILD_OSG_APPLICATIONS OFF CACHE INTERNAL "" )
    set( BUILD_OSG_EXAMPLES OFF CACHE INTERNAL "" )
    set( OSG_DETERMINE_WIN_VERSION   ON     CACHE INTERNAL "" )
    set( OSG_GL3_AVAILABLE   ON     CACHE INTERNAL "" )
    set( OSG_GL1_AVAILABLE   OFF    CACHE INTERNAL "" )
    set( OSG_GL2_AVAILABLE   OFF    CACHE INTERNAL "" )
    set( OSG_GLES1_AVAILABLE OFF    CACHE INTERNAL "" )
    set( OSG_GLES2_AVAILABLE OFF    CACHE INTERNAL "" )
    set( OSG_GL_DISPLAYLISTS_AVAILABLE OFF    CACHE INTERNAL "" )
    set( OSG_GL_MATRICES_AVAILABLE OFF    CACHE INTERNAL "" )
    set( OSG_GL_VERTEX_FUNCS_AVAILABLE OFF    CACHE INTERNAL "" )
    set( OSG_GL_VERTEX_ARRAY_FUNCS_AVAILABLE OFF    CACHE INTERNAL "" )
    set( OSG_GL_FIXED_FUNCTION_AVAILABLE OFF    CACHE INTERNAL "" )
    set( OSG_USE_LOCAL_LUA_SOURCE OFF CACHE INTERNAL "" )
    
    
I think the problem is because of  OSG_GL_FIXED_FUNCTION_AVAILABLE but I don't want to change my CMakeLists. How can I set size on points without using osg::Point->setSize?

**Comment by robertosfield** [comment, marked as answer]
Thankyou for providing further information which is crucial.  When you select to build with OpenGL fixed function switched off you can't use the GL fixed function state.

Do you actually need to compile without OpenGL fixed function backwards compatibility?  You loose a great deal of OSG functionality by disabling compatibility.

If you are running without fixed function pipeline compatibility then you'll need to provide your own uniforms to the shaders you write.

If you are knew to the OSG then I would recommend just moving the [VulkanSceneGraph](https://github.com/vsg-dev/VulkanSceneGraph).  The VSG is the OSG successor, it supports modern hardware and full shader usage far batter than OSG/OpenGL can ever do and will likely save you time as well as leave you with a better performing and easy to maintain software. 

 The OSG is a legacy API built upon a OpenGL which is a legacy graphics API, it's best kept for maintaining existing applications.
